File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/sql/MethodTest.java
package com.raizlabs.android.dbflow.test.sql;

import com.raizlabs.android.dbflow.sql.SQLiteType;
import com.raizlabs.android.dbflow.sql.language.Method;
import com.raizlabs.android.dbflow.test.FlowTestCase;

import org.junit.Test;

import static com.raizlabs.android.dbflow.sql.language.Method.avg;
import static com.raizlabs.android.dbflow.sql.language.Method.cast;
import static com.raizlabs.android.dbflow.sql.language.Method.count;
import static com.raizlabs.android.dbflow.sql.language.Method.group_concat;
import static com.raizlabs.android.dbflow.sql.language.Method.max;
import static com.raizlabs.android.dbflow.sql.language.Method.min;
import static com.raizlabs.android.dbflow.sql.language.Method.sum;
import static com.raizlabs.android.dbflow.sql.language.Method.total;
import static com.raizlabs.android.dbflow.test.structure.TestModel1_Table.name;
import static org.junit.Assert.assertEquals;

/**
 * Description: Tests a {@link Method} class.
 */
public class MethodTest extends FlowTestCase {

    @Test
    public void test_avgMethod() {
        String query = avg(name).getQuery();
        assertEquals(query, "AVG(`name`)");
    }

    @Test
    public void test_countMethod() {
        String query = count(name).getQuery();
        assertEquals(query, "COUNT(`name`)");
    }

    @Test
    public void test_groupConcatMethod() {
        String query = group_concat(name).getQuery();
        assertEquals(query, "GROUP_CONCAT(`name`)");
    }

    @Test
    public void test_maxMethod() {
        String query = max(name).getQuery();
        assertEquals(query, "MAX(`name`)");
    }

    @Test
    public void test_minMethod() {
        String query = min(name).getQuery();
        assertEquals(query, "MIN(`name`)");
    }

    @Test
    public void test_sumMethod() {
        String query = sum(name).getQuery();
        assertEquals(query, "SUM(`name`)");
    }

    @Test
    public void test_totalMethod() {
        String query = total(name).getQuery();
        assertEquals(query, "TOTAL(`name`)");
    }

    @Test
    public void test_castMethod() {
        String query = cast(name).as(SQLiteType.INTEGER).getQuery();
        assertEquals("CAST(`name` AS INTEGER)", query);
    }
}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/sql/MethodTest.kt
package com.raizlabs.android.dbflow.test.sql

import com.raizlabs.android.dbflow.sql.SQLiteType
import com.raizlabs.android.dbflow.sql.language.Method
import com.raizlabs.android.dbflow.test.FlowTestCase

import org.junit.Test

import com.raizlabs.android.dbflow.sql.language.Method.avg
import com.raizlabs.android.dbflow.sql.language.Method.cast
import com.raizlabs.android.dbflow.sql.language.Method.count
import com.raizlabs.android.dbflow.sql.language.Method.group_concat
import com.raizlabs.android.dbflow.sql.language.Method.max
import com.raizlabs.android.dbflow.sql.language.Method.min
import com.raizlabs.android.dbflow.sql.language.Method.replace
import com.raizlabs.android.dbflow.sql.language.Method.sum
import com.raizlabs.android.dbflow.sql.language.Method.total
import com.raizlabs.android.dbflow.test.structure.TestModel1_Table.name
import org.junit.Assert.assertEquals

/**
 * Description: Tests a [Method] class.
 */
class MethodTest : FlowTestCase() {

    @Test
    fun test_avgMethod() {
        val query = avg(name).query
        assertEquals(query, "AVG(`name`)")
    }

    @Test
    fun test_countMethod() {
        val query = count(name).query
        assertEquals(query, "COUNT(`name`)")
    }

    @Test
    fun test_groupConcatMethod() {
        val query = group_concat(name).query
        assertEquals(query, "GROUP_CONCAT(`name`)")
    }

    @Test
    fun test_maxMethod() {
        val query = max(name).query
        assertEquals(query, "MAX(`name`)")
    }

    @Test
    fun test_minMethod() {
        val query = min(name).query
        assertEquals(query, "MIN(`name`)")
    }

    @Test
    fun test_sumMethod() {
        val query = sum(name).query
        assertEquals(query, "SUM(`name`)")
    }

    @Test
    fun test_totalMethod() {
        val query = total(name).query
        assertEquals(query, "TOTAL(`name`)")
    }

    @Test
    fun test_castMethod() {
        val query = cast(name).`as`(SQLiteType.INTEGER).query
        assertEquals("CAST(`name` AS INTEGER)", query)
    }

    @Test
    fun test_replaceMethod() {
        val query = replace(name, "ro", "or").query
        assertEquals("REPLACE(`name` , 'ro' , 'or')", query)
    }
}
